test/Libraries/DynamoPythonTests/PythonEditTests.cs

```diff
@@ -42,6 +68,37 @@ public void PythonScriptEdit_WorkspaceChangesReflected()
             Assert.IsTrue(model.CurrentWorkspace.HasUnsavedChanges);
         }
 
+        [Test]
+        public void PythonNodeEnginePropertyTest()
+        {
+            // open file
+            var model = ViewModel.Model;
+            var examplePath = Path.Combine(TestDirectory, @"core\python", "python.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            // get the python node and check Engine property
+            var workspace = model.CurrentWorkspace;
+            var nodeModel = workspace.NodeFromWorkspace("3bcad14e-d086-4278-9e08-ed2759ef92f3");
+            var pynode = nodeModel as PythonNode;
+            Assert.AreEqual(pynode.Engine, PythonEngineVersion.IronPython2);
+
+            // workspace has no changes
+            Assert.IsFalse(model.CurrentWorkspace.HasUnsavedChanges);
+
+            // Serialize DYN and deserialize to double check check Engine field
+            var path = GetNewFileNameOnTempPath();
+            ViewModel.Model.CurrentWorkspace.Save(path);
+
+            var fileContents = File.ReadAllText(path);
+            if (string.IsNullOrWhiteSpace(fileContents))
+                return;
+
+            JObject dynObj = JObject.Parse(fileContents);
+            var pythonTokens = dynObj["Nodes"].Where(t => t.Value<string>("NodeType") == "PythonScriptNode").Select(t => t);
+            Assert.IsNotNull(pythonTokens);
+            Assert.IsTrue(pythonTokens.Any(t => t.Value<string>("Engine") == PythonEngineVersion.IronPython2.ToString()));
+        }
```

Comment:
How about if you changed the value to `CPython3` before saving? This would serve to check that the property can serialize both values properly.

Reply:
This test was added by Aaron to check serialization and deserialization of the Engine property on global level. There is another test PythonNodeEnginePropertyChangeTest that is added to assert the python node values for both the engines.

Reply:
I saw that test but it's not quite the same thing. My point regarding this suggestion is that it could catch a bug on serialization where the `Engine` of a python node would always be set to `IronPython2`. Not saying that's a bug we have or that it is likely to ever appear, but I just thought it would make the test more interesting to change the engine before serializing.